### PR TITLE
 New source md files, new logic for links in README

### LINKED ISSUE #36: Port over license info from whosonfirst-data/LICENSE.md
We note the equivalent license "code" (e.g.: `CC-BY`) and provide a copy of the actual license text in the [whosonfirst-data/LICENSE.md](https://github.com/whosonfirst-data/whosonfirst-data/blob/master/LICENSE.md) file. Let's move that info here (and point to this repo instead of duplicating the info).

For instance:

* `gn` - [GeoNames](http://www.geonames.org) (CC-BY)
* `gp` - [GeoPlanet](https://developer.yahoo.com/geo/geoplanet/) Where on Earth, aka `woe` (CC-BY)
* `ne` - [Natural Earth](http://www.naturalearthdata.com) (Public Domain)
* `oa` - [OurAirports](http://ourairports.com/) (Public Domain)
* `qs` - [Quattroshapes](https://github.com/foursquare/quattroshapes/blob/master/) (CC-BY)
* `zs` - [Zetashapes](http://zetashapes.com/) (CC-BY)
* `meso` - [Mesoshapes](https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/meso.json) (CC0)

And for **GeoNames**:

GeoNames database is licenced as follows:

```
The GeoNames geographical database is available for download free of
charge under a creative commons attribution license. 
```

### LINKED ISSUE #55: Reconcile WOF license page with sources page
Reconcile the [WOF license](https://github.com/whosonfirst-data/whosonfirst-data/blob/master/LICENSE.md) page with the [WOF sources](https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/README.md) README file. The information on the sources page is machine-generated, via the makefile, and updated each time a new source is added to the sources repo. The WOF license page, though, is a static page and not updated each time a source is added to Who's On First. 

Each time a source is added to the `whosonfirst-sources` repo, it should also be added to the list of sources in the README file.

### PR DESCRIPTION
**Pull request by stepps00**
(Replaces https://github.com/whosonfirst/whosonfirst-sources/pull/57)

This PR fixes #55 and fixes #36. Updates include:

- New .md files in the `/sources` folder for each source listed in the [Who's On Frist LICENSE.md file](https://github.com/whosonfirst-data/whosonfirst-data/blob/master/LICENSE.md)
- Each new .md file is now linked in the associated .json file's `"description"` property
- Updated the [docs.py script](https://github.com/whosonfirst/whosonfirst-sources/blob/master/bin/docs.py) with the following, which now results in links to each source instead of markdown text:

```
        if details.get('url'):
            docs.write("* %s _%s_\n" % ('url', details['url']))
        
        if details.get('license') and details.get('license').startswith("http"):
            docs.write("* %s _%s_\n" % ('license', details['license']))
        else:
            docs.write("* %s `%s`\n" % ('license', details['license']))
```

- Ran the makefile to update all spec docs and the README.md file (updated version with links [here](https://github.com/whosonfirst/whosonfirst-sources/blob/367f5266d7e6453a4dc1acb611148cdf3ad5d44c/sources/README.md))

### CODE REVIEW

File: sources/geoplanet.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
 	"fullname": "Yahoo! GeoPlanet",
 	"name": "geoplanet",
 	"prefix": "gp",
-    	"key": "id",
+    "key": "id",
```
Comment (nvkelso) [inline sources/geoplanet.json]: Match the spacing with other rows.

File: sources/geoplanet.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
...
 	"url": "http://developer.yahoo.com/geo/geoplanet/",
-	"license": "http://developer.yahoo.com/geo/geoplanet/data/"
+	"license": "http://developer.yahoo.com/geo/geoplanet/data/",
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/geoplanet.md"
```
Comment (nvkelso) [inline sources/geoplanet.json]: Match the spacing with other rows.

File: sources/meso.json [hunk 3]
```diff
@@ -6,5 +6,5 @@
 	"key":"",
 	"url": "https://github.com/whosonfirst-data/whosonfirst-data/blob/master/LICENSE.md",
 	"license": "https://github.com/whosonfirst-data/whosonfirst-data/blob/master/LICENSE.md",
-	"description": "Mesoshapes are a product of Who's On First under CC0."
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/meso.md"
```
Comment (nvkelso) [inline sources/meso.json]: Match the spacing with other rows.

File: sources/geonames.json [hunk 4]
```diff
@@ -3,7 +3,8 @@
 	"fullname": "GeoNames",
 	"name": "geonames",
 	"prefix": "gn",
-    	"key": "id",
+	"key": "id",
 	"url": "http://www.geonames.org/",
-	"license": "http://www.geonames.org/about.html"
+	"license": "http://www.geonames.org/about.html",
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/geonames.md"
```
Comment (nvkelso) [inline sources/geonames.json]: Match the spacing with other rows.

File: sources/naturalearth.json [hunk 5]
```diff
@@ -5,5 +5,6 @@
 	"prefix": "ne",
 	"key":"id",
 	"url": "http://www.naturalearthdata.com/",
-	"license": "http://www.naturalearthdata.com/about/terms-of-use/"
+	"license": "http://www.naturalearthdata.com/about/terms-of-use/",
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/naturalearth.md"
```
Comment (nvkelso) [inline sources/naturalearth.json]: Match the spacing with other rows.

File: sources/os.json [hunk 6]
```diff
@@ -5,5 +5,6 @@
 	"prefix": "os",
 	"key": "",
 	"url": "https://www.ordnancesurvey.co.uk",
-	"license": "https://www.ordnancesurvey.co.uk/business-and-government/licensing/using-creating-data-with-os-products/os-opendata.html"
+	"license": "https://www.ordnancesurvey.co.uk/business-and-government/licensing/using-creating-data-with-os-products/os-opendata.html",
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/os.md"
```
Comment (nvkelso) [inline sources/os.json]: Match the spacing with other rows.

File: sources/ourairports.json [hunk 7]
```diff
@@ -3,7 +3,8 @@
 	"fullname": "OurAirports",
 	"name": "ourairports",
 	"prefix": "oa",
-    	"key": "id",
+    "key": "id",
```
Comment (nvkelso) [inline sources/ourairports.json]: Match the spacing with other rows.

File: sources/ourairports.json [hunk 8]
```diff
@@ -3,7 +3,8 @@
...
 	"url": "http://ourairports.com/data/",
-	"license": "http://ourairports.com/"
+	"license": "http://ourairports.com/",
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/ourairports.md"
```
Comment (nvkelso) [inline sources/ourairports.json]: Match the spacing with other rows.

File: sources/quattroshapes.json [hunk 9]
```diff
@@ -5,5 +5,6 @@
 	"prefix": "qs",
    	"key": "id",
 	"url": "http://www.quattroshapes.com/",
-	"license": "https://github.com/foursquare/quattroshapes/blob/master/LICENSE.md"
+	"license": "https://github.com/foursquare/quattroshapes/blob/master/LICENSE.md",
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/quattroshapes.md"
```
Comment (nvkelso) [inline sources/quattroshapes.json]: Match the spacing with other rows.

File: sources/zetashapes.json [hunk 10]
```diff
@@ -5,5 +5,6 @@
 	"prefix": "zs",
 	"key":"id",
 	"url": "http://www.zetashapes.com",
-	"license": "http://www.zetashapes.com/license"
+	"license": "http://www.zetashapes.com/license",
+    "description": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/zetashapes.md"
```
Comment (nvkelso) [inline sources/zetashapes.json]: Match the spacing with other rows.

**Comment by thisisaaronland** [comment]
I would like to keep a single inline description in the machine readable source blob. That way there is always a description at hand without having to load – and importantly parse – a separate markdown file.

The purpose of the markdown is to get in to the details of a thing, which is supplemental to a short description.

I would propose keeping `description` as a string, and `details` as a pointer to a "free form" markdown file.

**Comment by nvkelso** [comment]
I like that approach. Here's an existing example with a string `description`: 

- https://github.com/whosonfirst/whosonfirst-sources/blob/8e24679b56124e65edd75f68c980fe380b903e44/sources/simplegeo.json

And for mesoshapes it'd look like:

```
{
	"id": 1108756907,
	"fullname": "Mesoshapes",
	"name": "meso",
	"prefix": "meso",
	"key":"",
	"url": "https://github.com/whosonfirst-data/whosonfirst-data/blob/master/LICENSE.md",
	"license": "https://github.com/whosonfirst-data/whosonfirst-data/blob/master/LICENSE.md",
	"description": "Mesoshapes are a product of Who's On First under CC0.",
        "details": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/meso.md"
}
```

**Comment by stepps00** [comment]
Looks good. After discussion, we've settled on `"remarks"`, instead of `"details"`, which results in something like this for the Ordnance Survey:

```
{
	"id": 874390485,
	"fullname": "Ordnance Survey",
	"name": "ordnancesurvey",
	"prefix": "os",
	"key": "",
	"url": "https://www.ordnancesurvey.co.uk",
	"license": "https://www.ordnancesurvey.co.uk/business-and-government/licensing/using-creating-data-with-os-products/os-opendata.html",
	"description": "A non-ministerial government agency that acts as the national mapping agency for Great Britain.",
	"remarks": "https://github.com/whosonfirst/whosonfirst-sources/blob/master/sources/os_remarks.md"
}

```

Note the markdown filename.

**Comment by stepps00** [comment]
@nvkelso - newest commit includes:
 
- Fixes for all spacing issues
- New `"remarks"` property in sources with an .md file
- Appends `_remarks` to each .md filename
- Logic to handle new `remarks` property in the `docs.py` script

File: sources/quattroshapes.json [hunk 11]
```diff
@@ -5,5 +5,7 @@
 	"prefix": "qs",
    	"key": "id",
 	"url": "http://www.quattroshapes.com/",
-	"license": "https://github.com/foursquare/quattroshapes/blob/master/LICENSE.md"
+	"license": "https://github.com/foursquare/quattroshapes/blob/master/LICENSE.md",
+	"description": "A gazetteer of non-overlapping, authoritative polygons around a curated list of places.",
```
Comment (nvkelso) [inline sources/quattroshapes.json]: For the description here please add: 

`Quattroshapes includes open data from government and other sources, many of which require attribution. See [remarks](quattroshapes_remarks.md) for details.`

**Pull request merged**